jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/JmxConnectionTest.java

```diff
@@ -191,11 +192,11 @@ private static void checkConnectionLogs(JmxScraperContainer scraper, boolean exp
   }
 
   private static void waitTerminated(GenericContainer<?> container) {
-    int retries = 10;
+    long retries = TimeUnit.SECONDS.toMillis(10);
```

Comment:
This is not a duration here, just a retry count.

```suggestion
    int retries = 10;
```

Reply:
The fact that you were treating it as a count and not a time is exactly what was causing it to fail. It wasn't waiting long enough.

I hope we can all agree that a fixed duration over a maximum number of iterations still has a maximum time. I understood that the only reason it looped at all was to prevent waiting unnecessarily long.